pilot/utils/llm_connection.py

```diff
@@ -96,10 +96,14 @@ def create_gpt_chat_completion(messages: List[dict], req_type, project,
                 del gpt_data[key]
 
     # Advise the LLM of the JSON response schema we are expecting
+    messages_length = len(gpt_data['messages'])
     add_function_calls_to_request(gpt_data, function_calls)
 
     try:
         response = stream_gpt_completion(gpt_data, req_type, project)
+
+        # Remove JSON schema and any added retry messages
+        gpt_data['messages'] = gpt_data['messages'][:messages_length]
```

Comment:
can you explain what exactly is this doing and why?

Reply:
Modifying `gpt_data['messages']` didn't actually do anything, so I've changed it to pop the new messages from the list.

`add_function_calls_to_request()` and `@retry_on_exception` add extra messages to the conversation (JSON schema and validation errors) which do not add value to the context in subsequent completion requests